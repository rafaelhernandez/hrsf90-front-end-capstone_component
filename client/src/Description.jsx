import React from 'react';
import Header from './Header.jsx';
import Summary from './Summary.jsx';
import SpaceDescription from './SpaceDescription.jsx';
import ContactHost from './ContactHost.jsx';
import Amenities from './Amenities.jsx';
import HouseRules from './HouseRules.jsx';
import Cancellations from './Cancellations.jsx';

const Description = (props) => (
  <div className="div--description-description">
    <Header desc={props.desc.description}></Header>
    <Summary summary={props.desc.description.summary}></Summary>
    <SpaceDescription desc={props.desc.description.description}></SpaceDescription>
    <ContactHost></ContactHost>
    <Amenities amenities={props.desc.description.amenities}></Amenities>
    <HouseRules desc={props.desc.description}></HouseRules>
    <Cancellations cancellationPolicy={props.desc.description.cancellation_policy}></Cancellations>
  </div>
);

export default Description;
