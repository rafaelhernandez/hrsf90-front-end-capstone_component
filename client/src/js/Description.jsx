import React from 'react';
import Header from './Header.jsx';
import Summary from './Summary.jsx';
import SpaceDescription from './SpaceDescription.jsx';
import ContactHost from './ContactHost.jsx';
import Amenities from './Amenities.jsx';
import HouseRules from './HouseRules.jsx';
import Cancellations from './Cancellations.jsx';
import Separator from './Separator.jsx';

const Description = (props) => (
  <div className="div-description-description">
    <div className="div-description-body">
      <Header desc={props.desc.description}></Header>
      <Summary summary={props.desc.description.summary}></Summary>
      <SpaceDescription desc={props.desc.description}></SpaceDescription>
      <ContactHost></ContactHost>
      <Separator></Separator>
      <Amenities amenities={props.desc.description.amenities}></Amenities>
      <Separator></Separator>
      <HouseRules desc={props.desc.description}></HouseRules>
      <Separator></Separator>
      <Cancellations cancellationPolicy={props.desc.description.cancellation_policy}></Cancellations>
      <Separator></Separator>
    </div>
  </div>
);

export default Description;
