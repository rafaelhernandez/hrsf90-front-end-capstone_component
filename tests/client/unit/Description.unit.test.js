import React from 'react';
import { configure } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import { shallow } from 'enzyme';
import { shallowToJson } from 'enzyme-to-json';
import Description from '../../../client/src/Description.jsx';
import descriptionSample from '../../../db/data/description_sample.js';

configure({ adapter: new Adapter() });

describe('Description', () => {
  it('should render correctly', () => {
    const output = shallow(
      <Description desc={descriptionSample}></Description>
    );
    expect(shallowToJson(output)).toMatchSnapshot();
  });
});
