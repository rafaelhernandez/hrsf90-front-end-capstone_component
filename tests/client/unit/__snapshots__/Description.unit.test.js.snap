// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Description should render correctly 1`] = `
<div
  className="div--description-description"
>
  <Header
    desc="One bedroom sanctuary with a beautiful garden view and cozy fireplace. Comes with it's own entrance just steps from bus lines, train and all that Haight Street has to offer. Complete privacy with a separate bedroom, bathroom, and  breakfast bar.

If you want to be within walking distance to Golden Gate Park, Haight St, Buena Vista Park, and Cole Valley, this is the spot. Full size bed in the bedroom with 400 thread count sheets and a pull-out twin in the living room. Just installed fireplace and large screen tv makes hanging out fun in the comfortable  living room. Bathroom has a shower within the tub and white fluffy towels.

Your own entrance means you can come and go as you wish and enter the apartment through new french doors from the garden. There is a mini fridge, Nespresso coffee maker, toaster oven and goodies awaiting you in the breakfast bar area. Free wifi, cable, and maps of the area as well. Laundry is  available upon request conveniently located just outside the door inside the apartment.

I live upstairs and will meet you upon arrival then be available as much as you need or not.  I've been exploring the neighborhood and take long walks with my dog so there's not much I haven't seen!

You will hear from many that this is the  best neighborhood in SF. I chose this area for the many  Victorian homes, the ease of mass transit, the walkability, and the abundance of parks,restaurants, and nightlife nearby.
As a recent transplant to this area I can tell you that the stores, coffee shops, restaurants, and pubs are great here. I forget I have a car most of the time! Street parking is easy except for Wednesday 8-10 due to street cleaning.

Everything is a walk away or jump on the many bus lines that are at the corner. In minutes you can be at the Ferry Building, Marina, Presidio, Castro, UCSF Medical building, Market Street and Union Square. The N-Judah line also stops in nearby Cole Valley. I walk most places - post office, grocery store, pharmacy, bank, hardware, and clothing stores. The Magnolia Brew Pub is a block away with all their own draft beers. The number of walking distance hang-outs is amazing - Zam Zam is a fun martini bar, Sparrow is my new favorite farm to table restaurant and Club Deluxe offers pizza and live music every night! Plus Golden Gate Park is a walk away which means you can walk to all the museums and gardens there. I have books, maps and all the info you'll need to navigate your trip here.

I live upstairs and will be available for any questions - my friendly Australian Shepherd, Boo,  may come downstairs to say hello in the backyard.
I have  registered with the city of San Francisco and my license is #STR- (Phone number hidden by Airbnb)"
  />
  <Summary />
  <SpaceDescription />
  <ContactHost />
  <Amenities />
  <HouseRules
    desc="One bedroom sanctuary with a beautiful garden view and cozy fireplace. Comes with it's own entrance just steps from bus lines, train and all that Haight Street has to offer. Complete privacy with a separate bedroom, bathroom, and  breakfast bar.

If you want to be within walking distance to Golden Gate Park, Haight St, Buena Vista Park, and Cole Valley, this is the spot. Full size bed in the bedroom with 400 thread count sheets and a pull-out twin in the living room. Just installed fireplace and large screen tv makes hanging out fun in the comfortable  living room. Bathroom has a shower within the tub and white fluffy towels.

Your own entrance means you can come and go as you wish and enter the apartment through new french doors from the garden. There is a mini fridge, Nespresso coffee maker, toaster oven and goodies awaiting you in the breakfast bar area. Free wifi, cable, and maps of the area as well. Laundry is  available upon request conveniently located just outside the door inside the apartment.

I live upstairs and will meet you upon arrival then be available as much as you need or not.  I've been exploring the neighborhood and take long walks with my dog so there's not much I haven't seen!

You will hear from many that this is the  best neighborhood in SF. I chose this area for the many  Victorian homes, the ease of mass transit, the walkability, and the abundance of parks,restaurants, and nightlife nearby.
As a recent transplant to this area I can tell you that the stores, coffee shops, restaurants, and pubs are great here. I forget I have a car most of the time! Street parking is easy except for Wednesday 8-10 due to street cleaning.

Everything is a walk away or jump on the many bus lines that are at the corner. In minutes you can be at the Ferry Building, Marina, Presidio, Castro, UCSF Medical building, Market Street and Union Square. The N-Judah line also stops in nearby Cole Valley. I walk most places - post office, grocery store, pharmacy, bank, hardware, and clothing stores. The Magnolia Brew Pub is a block away with all their own draft beers. The number of walking distance hang-outs is amazing - Zam Zam is a fun martini bar, Sparrow is my new favorite farm to table restaurant and Club Deluxe offers pizza and live music every night! Plus Golden Gate Park is a walk away which means you can walk to all the museums and gardens there. I have books, maps and all the info you'll need to navigate your trip here.

I live upstairs and will be available for any questions - my friendly Australian Shepherd, Boo,  may come downstairs to say hello in the backyard.
I have  registered with the city of San Francisco and my license is #STR- (Phone number hidden by Airbnb)"
  />
  <Cancellations />
</div>
`;
